code/modules/assembly/pass.dm

```diff
@@ -0,0 +1,36 @@
+/obj/item/assembly/timer/pass
+	name = "Personal Alert Safety System"
+	desc = "A device that plays a very loud alarm if it isn't moved for 30 seconds."
+	icon_state = "timer"
+	var/lastloc = null
+	var/loca = null
+	var/alarm = 0
+	var/list/pass_sounds = list('sound/effects/alert.ogg' = 1)
+
+
+
+/obj/item/assembly/timer/pass/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, pass_sounds, 100)
+
+
+///obj/item/assembly/timer/pass/describe()
+//	if(timing)
+//		return "The PASS will go off in [time] seconds if not moved!"
+//	return "The timer is set for [time] seconds."
+
+
+/obj/item/assembly/timer/pass/process()
+	if(alarm == 1)
+		playsound(loca,'sound/effects/alert.ogg', 150, 1)
+	if(timing && (time > 0))
+		loca = get_turf(src)
+		if(loca == lastloc)
+			time -= 2 // 2 seconds per process()
```

Comment:
Shouldn't it go down by 1 second so that way if you have the examine text it will be accurate?

Reply:
Ty. that is copy/pasted from the timer code. Soooo... for some reason, this is what works.

Reply:
Does it work though? The processing subsystem says it process every 1 second https://github.com/ParadiseSS13/Paradise/blob/0fb164d041aa7db36ea6fa432284559bac6aa250/code/controllers/subsystem/processing/processing.dm

Reply:
It is possible that the timer code is outdated

Reply:
I'm not sure. When I tested this locally I only really checked to see if 1) the alarm went off continuously when not moving for x seconds and 2) the alarm shut up when you moved